src/plugins/stdioproc/README.md

```diff
@@ -0,0 +1,201 @@
+- infos = Information about the stdioproc plugin is in keys below
+- infos/author = Klemens Böswirth <k.boeswirth+git@gmail.com>
+- infos/licence = BSD
+- infos/needs =
+- infos/provides =
+- infos/recommends =
+- infos/placements = postgetstorage presetstorage
+- infos/status = maintained unittest shelltest experimental
+- infos/metadata =
+- infos/description = one-line description of stdioproc
+
+## Introduction
+
+This plugin spawns a new process with a user-defined executable and delegates all operations to the new process.
+
+## Usage
+
+Set the config key `app` and the arrays `args/#`, `env/#` to the path of an executable, the arguments that shall be passed and the environment variables to be set.
+
+```
+kdb mount test.dump /tests/stdioproc stdioproc 'app=/usr/bin/pluginproc' 'args=#1' 'args/#0=--load-plugin' 'args/#1=myplugin'
+```
+
+During `elektraStdprocioOpen` the plugin will collect the `args/#` and `env/#` values into two arrays `argv` and `envp`.
+It then `fork`s a new process and the child calls `execve` with the path from `app` as well as `argv` and` envp`.
+The child process is expected to listen to `stdin` and write to `stdout` according to the protocol described below.
+
+If communication can be established, the child-process will be kept running until `elektraStdprocioClose`.
+
+## Protocol
+
+The entire protocol is text-based (apart from binary key values) and request-response-based, and happens over `stdin`/`stdout`.
+The parent process sends request to the child process.
+The child then processes the request and sends a response back.
+
+To encode keysets and keys we use the format of the `dump` plugin.
+It is important to note that the `dump` plugin is instructed to write the full keynames (normally it removes the parent prefix).
+There is, however, an exception during initialization, which is described in the appropriate section.
+The child-process may also delegate directly to the `dump` plugin, or re-implement the encoding.
+
+The communications protocol used by the plugin has 3 phases:
+
+- Initialization: Initial handshake after first starting the child-process.
+- Operation: The main phase, which implements all the operations of an Elektra plugin (open, get, set, close, ...).
+- Termination: The plugin is being closed and the parent-process tells the child to shut down.
+
+> **Note:** In the sections below we use the following format to describe messages:
+>
+> ```
+> Parent > Child
+>
+> HELLO WORLD
```

Comment:
protocol version here?

Reply:
This is just an example of how the protocol is described within the README. It is not part of the actual protocol.